# What is this website?

This website complements a published benchmark of polygenic score development methods across five European biobanks, available below:

> Remo Monti, Lisa Eick, Georgi Hudjashov, Kristi LÃ¤ll, Stavroula Kanoni, Brooke N Wolford, Benjamin Wingfield, Oliver Pain, Sophie Wharrie, Bradley Jermy, Aoife McMahon, Tuomo Hartonen, Henrike O Heyne, Nina Mars, Genes & Health Research Team, Kristian Hveem, Michael Inouye, David A van Heel, Reedik Magi, Pekka Marttinen, Samuli Ripatti, Andrea Ganna, Christoph Lippert. "Evaluation of polygenic scoring methods in five biobanks reveals greater variability between biobanks than between methods and highlights benefits of ensemble learning". medRxiv 2023. doi: https://doi.org/10.1101/2023.11.20.23298215

The analysis is complex, and the aim of the interactive visualisations on this website is to make the benchmark results clearer.

::: {.callout-tip title="Get started"}
Check out a [visualisation of effect sizes](effect-sizes.qmd) or compare [methods directly](pairwise.qmd)
:::

# What is a polygenic score?

From the [PGS Catalog](https://www.pgscatalog.org):

> "A polygenic score (PGS) aggregates the effects of many genetic variants into a single number which predicts genetic predisposition for a phenotype. PGS are typically composed of hundreds-to-millions of genetic variants (usually SNPs) which are combined using a weighted sum of allele dosages multiplied by their corresponding effect sizes, as estimated from a relevant genome-wide association study (GWAS)."

# How are polygenic scores made?

PGS are typically developed using large amounts of data from a biobank, like the [UK Biobank](https://www.ukbiobank.ac.uk). A biobank is a large medical repository, containing linked genetic and health information. Using a combination of genetic data and health information about disease outcomes, it's possible for researchers to identify genetic variants that contribute to disease risk. 

Making new polygenic scores is a complicated data science process that involves tuning hyperparameters and validating how well scores generalise on new data across different ancestry groups. Validated scores are often published in the [PGS Catalog](https://www.pgscatalog.org) for people to reuse.

# How do people use polygenic scores?

After a PGS is made the scoring file (a list of variants and effect weights) can be uploaded to the PGS Catalog for people to reuse. If you have some human genetic data it's possible to use published scoring files to predict genetic predisposition for traits. Reusing published scores is much simpler than making new PGS, especially if you use the [PGS Catalog Calculator](https://github.com/PGScatalog/pgsc_calc) ðŸ‘€

# About this website

## Who might find the website useful?

If you'd like to develop a new genetic score and you're not sure which method to use, the benchmarks we've published here could be helpful for you.

You won't find this website useful if you're browsing the PGS Catalog and are trying to understand which scoring file is best to use with your genetic data for particular phenotype. **We only assess and compare methods that can develop new scores**. 

## What work did you do?

* We evaluated scores from seven methods in five biobanks across 16 endpoints, [building on the work done by GenoPred](https://opain.github.io/GenoPred/)
* We conducted meta-analyses to quantify the effects of method choice, hyperparameter tuning, method ensembling and target biobank on PGS performance
* We published interactive visualisations here to summarise and explain the work described above


## Who made this website?

[INTERVENE](https://www.interveneproject.eu/who-we-are) is an international and interdisciplinary consortium that seeks to develop and implement next-generation tools for AI-facilitated personalised medicine.

The Genetic Data Platform team at [EMBL-EBI](https://www.ebi.ac.uk/) built this website as an INTERVENE partner with Remo Monti at the [Hasso Plattner Institut](https://hpi.de).

## Are the data available to download?

The data and website code are open source and permissively licensed:

* The data are published in an [R data package](https://github.com/intervene-EU-H2020/pgsCompaR). Have fun! 
* The website is [built with Quarto, R, and Observable JS](https://github.com/intervene-EU-H2020/pgs-method-compare)

## How do I get help?

If something isn't clear or you have questions, please feel free [to open an issue on Github.](https://github.com/intervene-EU-H2020/pgs-method-compare/issues/new)
